iree/tools/iree-benchmark-module-main.cc

```diff
@@ -230,7 +201,37 @@ int main(int argc, char** argv) {
       absl::flags_internal::OnUndefinedFlag::kIgnoreUndefined);
   ::benchmark::Initialize(&argc, argv);
   iree::InitializeEnvironment(&argc, &argv);
-  IREE_CHECK_OK(iree::RegisterModuleBenchmarks());
+
+  iree_vm_instance_t* instance = nullptr;
+  iree_hal_device_t* device = nullptr;
+  iree_vm_module_t* hal_module = nullptr;
+  iree_vm_context_t* context = nullptr;
+  iree_vm_module_t* input_module = nullptr;
+  iree_vm_function_t function;
+  iree::vm::ref<iree_vm_list_t> inputs;
+  std::vector<iree::RawSignatureParser::Description> output_descs;
+  std::string module_data;
+  auto function_name = absl::GetFlag(FLAGS_entry_function);
+
+  // Capture status checking will be delayed at benchmarking runtime.
+  auto module_data_status = iree::GetModuleContentsFromFlags(module_data);
+  auto prepare_vm_func_status = iree::PrepareIREEVMFunction(
+      instance, device, hal_module, &context, input_module, function, inputs,
+      output_descs, function_name, module_data);
+
+  // Register function benchmarks...
+  iree::RegisterModuleBenchmarks(module_data_status, prepare_vm_func_status,
```

Comment:
This is super weird... You're passing the status to another function so that that function can do a check on it?

Reply:
If we checked the status here we will fail early and something like `list-benchmarks` won't work

Reply:
Ah ok. I think that's worth a comment.

Also why do these have to go from StatusOr to Status with an output arg? Can we not thread through the StatusOr in the same way?